roachtest: collect heap profiles

tbg opened the pull request:
Thanks to https://github.com/cockroachdb/roachprod/pull/83, we can
collect the logs for local clusters like we do for remote ones,
addressing a TODO.

Allow passing arguments to `roachprod` during start. This highlight
awkward problems with the separation of `roachtest/roachprod` and in
particular the distance to the actual `cockroach start` invocation.
However, not a thread I'm willing to pull now.

Pass an env var in the drop test that enables collection of heap
profiles (which are then collected as artifacts on teardown).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/22972)
<!-- Reviewable:end -->

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/22972#-:-L5zWEOEtJe8Q9rHhVb2:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tbg committed:
roachtest: collect heap profiles

Thanks to https://github.com/cockroachdb/roachprod/pull/83, we can
collect the logs for local clusters like we do for remote ones,
addressing a TODO.

Allow passing arguments to `roachprod` during start. This highlight
awkward problems with the separation of `roachtest/roachprod` and in
particular the distance to the actual `cockroach start` invocation.
However, not a thread I'm willing to pull now.

Pass an env var in the drop test that enables collection of heap
profiles (which are then collected as artifacts on teardown).

Release note: None

tschottdorf commented:
TFTR!

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/22972#-:-L60IU_OptsFFwz0Fayv:b-814g8k)*
<!-- Sent from Reviewable.io -->
